WalletWasabi/Extensions/LinqExtensions.cs

```diff
@@ -83,6 +83,26 @@ public static bool NotNullAndNotEmpty<T>(this IEnumerable<T> source)
 			return !(source is null) && source.Any();
 		}
 
+		public static IEnumerable<IEnumerable<T>> CombinationsWithoutRepetition<T>(
+			this IEnumerable<T> items,
+			int ofLength)
+		{
+			return (ofLength == 1) ?
+				items.Select(item => new[] { item }) :
+				items.SelectMany((item, i) => items.Skip(i + 1)
+												.CombinationsWithoutRepetition(ofLength - 1)
+												.Select(result => new T[] { item }.Concat(result)));
+		}
+
+		public static IEnumerable<IEnumerable<T>> CombinationsWithoutRepetition<T>(
+			this IEnumerable<T> items,
+			int ofLength,
+			int upToLength)
+		{
+			return Enumerable.Range(ofLength, Math.Max(0, upToLength - ofLength + 1))
+							.SelectMany(len => items.CombinationsWithoutRepetition(ofLength: len));
+		}
+
```

Comment:
Can you write tests for these? These're complex algorithms, I'd be able to understand it better from tests.

Reply:
Sure. It is a good idea.

Reply:
Done.